### PR TITLE
Modify configs to support grpc-netty-shaded

### PR DESCRIPTION
**Pull request by dzou**
Modify configurations to support `grpc-netty-shaded`.

The pom file for the sample is now a lot more simplified: https://github.com/GoogleCloudPlatform/google-cloud-graalvm-support/compare/use-netty-shaded?expand=1#diff-bd6c1dc3b6eccc9a9c14673eee04261e

I tried to support both non-shaded and shaded versions of `grpc-netty` but it was a bit too complicated for now. I think the first iteration of the project should just support `grpc-netty-shaded` which is what is inherited by the client libraries.

### CODE REVIEW

**Review by meltsufin — APPROVED** [review]
The goal is to support client libraries. So, I think it's a good call to focus on `grpc-netty-shaded`.

File: google-cloud-graalvm-samples/google-cloud-graalvm-pubsub-sample/pom.xml [hunk 1]
```diff
@@ -31,41 +28,23 @@
...
-        </exclusion>
-      </exclusions>
     </dependency>
 
     <dependency>
       <groupId>io.grpc</groupId>
-      <artifactId>grpc-netty</artifactId>
+      <artifactId>grpc-netty-shaded</artifactId>
+      <!-- Remove this dep after libraries-bom is updated to offer this version -->
+      <version>1.32.1</version>
```
Comment (elefeint) [inline google-cloud-graalvm-samples/google-cloud-graalvm-pubsub-sample/pom.xml:37]: Should this go into dependencyManagement instead?

**Review by elefeint — CHANGES_REQUESTED** [review]
Version override should probably go into dependency management.

File: google-cloud-graalvm-samples/google-cloud-graalvm-pubsub-sample/pom.xml (hunk 1 above)
Comment (dzou) [reply google-cloud-graalvm-samples/google-cloud-graalvm-pubsub-sample/pom.xml]: Done.

**Review by elefeint — APPROVED** [review]

**Pull request merged**